src/sage/rings/polynomial/generalized_order.py

```diff
@@ -0,0 +1,428 @@
+        return tuple(vector(self.greatest_tuple(*[tuple(t + l) for l in L])) - t)
+              
+    def is_in_cone(self,i,t): 
+        r"""
+        Test whether the tuple ``t`` is contained in the ``i``-th cone or not.
+
+        INPUT:
+
+        - ``i`` -- an integer, a cone index
+        - ``t`` -- a tuple of integers
+
+        EXAMPLES::
+            
+            sage: from sage.rings.polynomial.generalized_order import GeneralizedOrder
+            sage: order = GeneralizedOrder(2)
+            sage: order.is_in_cone(0, (1,2)) 
+            True
+            sage: order.is_in_cone(0,(-2,3))
+            False
+        """
+        return all(c >=0 for c in self.cone(i)*vector(t))
+
+    def generator(self,i,L):
+        r"""
+        Return the generator of the module over the ``i``-th cone for ``L``.
+
+        This is the monoïd of elements `t \in \ZZ^n` such that the greatest 
+        tuple of `t + L` is contained in the ``i``-th cone.
+
+        INPUT:
+            
+        - ``i`` -- an integer, a cone index
+        - ``L`` -- a list of iinteger tuples
+
+        EXAMPLES::
+
+            sage: from sage.rings.polynomial.generalized_order import GeneralizedOrder
+            sage: order = GeneralizedOrder(3)
+            sage: L = [(1,2,-2), (3,4,-5), (-6,2,-7), (3,-6,1)]
+            sage: order.generator(0,L)
+            (6, 6, 7)
+
+            sage: order.generator(2,L)
+            (5, 5, 6)
+        """
+    
+        cone_matrix = self._cones[i]
+        t = vector(self.translate_to_cone(i,L))
+        L = [vector(l) for l in L]
+        for c in cone_matrix.columns():
+            while self.is_in_cone(i,self.greatest_tuple(*[tuple(t + l) for l in L])):
+                t = t - c
+            t = t + c
+        return tuple(t)
+
+    def generator_for_pair(self,i,L1,L2):
+        r"""
+        Return the generator of the module over the ``i``-th cone for ``L1`` and ``L2``. 
+
+        INPUT:
+
+        - ``i`` -- an integer, a cone index
+        - ``L1`` -- a list of integer tuples
+        - ``L2`` -- a list of integer tuples
+
+        EXAMPLES::
+
+            sage: from sage.rings.polynomial.generalized_order import GeneralizedOrder
+            sage: order = GeneralizedOrder(2)
+            sage: L1 = [(2,3),(-4,2),(-1,-2)]
+            sage: L2 = [(1,-6),(5,-2),(3,4)]
+            sage: order.generator_for_pair(1,L1,L2)
+            (1, 5)
+
+        """
+        cone_matrix = self._cones[i]
+        lm1 = vector(self.greatest_tuple_for_cone(i,*L1))
+        lm2 = vector(self.greatest_tuple_for_cone(i,*L2))
+        g1 = vector(self.generator(i,L1))
+        g2 = vector(self.generator(i,L2))
+        m = vector([max(a,b) for a,b in zip(lm1 + g1, lm2 + g2)])
+        return tuple(cone_matrix*m)
+   
+# Score functions. Add more here and update the __init__ method.
+def min_score_function(t):
+        return -min(0,*t)
+
+def degmin_score_function(t):
+        return sum(t) -(len(t)+1)*min(0,*t)
```

Comment:
These two functions should be documented.
You also have a problem with identation.